### ISSUE TITLE
Intermitent issue when generating podcasts with Gemini

### ISSUE THREAD

**Issue by lfnovo** [souzatharsis/podcastfy#162]
[transcript_dd5be1335f49402a9ede53a52e24a802.txt](https://github.com/user-attachments/files/17670174/transcript_dd5be1335f49402a9ede53a52e24a802.txt)

Hey, sometimes when I try to generate a gemini based podcast, I end up with an error like this. Probably need to do some length checking on the markup. Attached the transcript for debugging.

```
  File "/Users/luisnovo/dev/projetos/open-notebook/open_notebook/plugins/podcasts.py", line 74, in generate_episode
    audio_file = generate_podcast(
                 ^^^^^^^^^^^^^^^^^
  File "/Users/luisnovo/dev/projetos/open-notebook/.venv/lib/python3.11/site-packages/podcastfy/client.py", line 340, in generate_podcast
    return process_content(
           ^^^^^^^^^^^^^^^^
  File "/Users/luisnovo/dev/projetos/open-notebook/.venv/lib/python3.11/site-packages/podcastfy/client.py", line 117, in process_content
    text_to_speech.convert_to_speech(qa_content, audio_file)
  File "/Users/luisnovo/dev/projetos/open-notebook/.venv/lib/python3.11/site-packages/podcastfy/text_to_speech.py", line 103, in convert_to_speech
    audio_data = self.provider.generate_audio(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/luisnovo/dev/projetos/open-notebook/.venv/lib/python3.11/site-packages/podcastfy/tts/providers/gemini.py", line 101, in generate_audio
    raise RuntimeError(f"Failed to generate audio: {str(e)}") from e
RuntimeError: Failed to generate audio: 400 This request's MultiSpeakerMarkup is too long.
```

**Comment by souzatharsis** [comment]
Each speaker turn has a max supported length by Google TTS Multispeaker
model. There are two challenges here:
1. Google does not disclose the max supported length
2. Gemini does not reliably follow the instruction pertaining to max number
of characters per turn

I am open to suggestions on how to solve this problem!
<http://linkedin.com/in/tharsissouza>

**Comment by souzatharsis** [comment]
I'll prioritize this as highest priority along with long-form podcast generation. https://github.com/souzatharsis/podcastfy/issues/166

**Comment by souzatharsis** [comment]
In addition to max length limit per turn, Multispeaker model has a 5000 bytes max limit for input text.
I'll try to optimally split input text and text per turn before doing text synthesis as a solution.

Additionally, integrating with Journey voices might be a compromise they are of high quality and way cheaper.

**Comment by souzatharsis** [comment]
This has been solved in release v0.4.0.

- All sample audio in README have been updated to use the new TTS Model. Added some longform podcasts too.

- I've updated python notebook describing longform podcast + new Google TTS model work:

https://github.com/souzatharsis/podcastfy/blob/main/podcastfy.ipynb
